### PR TITLE
Add TAP test (ipv4 broadcast)

### PR DESCRIPTION
**Pull request by songgao**
I added some test to make sure the `tap` interface is working. @hsheth2 @sam-falvo Could you guys take a look when you have time? Thanks!

Unfortunately Travis-CI [doesn't seem to support `/dev/net/tun`](https://github.com/travis-ci/travis-ci/issues/1503) :(

### CODE REVIEW

File: ipv4_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	setupIfce(t, net.IPNet{IP: self, Mask: mask}, ifce.Name())
+	startBroadcast(t, brd)
+
+	dataCh := make(chan []byte, 8)
+	startRead(dataCh, ifce)
+
+	timeout := make(chan bool, 1)
+	startTimeout(timeout, time.Second*8)
```
Comment (hsheth2) [inline ipv4_test.go]: It may be simpler to use `time.Timer` rather than using a custom timer goroutine.

**Comment by hsheth2** [comment]
The test does seem to work on my linux machine. However, on non-linux machines, the test will call `NewTap` before `setupIfce`, and so the test will panic rather than failing gracefully.

File: ipv4_test.go (hunk 1 above)
Comment (songgao) [inline ipv4_test.go]: Sounds good! I'll update it

**Comment by songgao** [comment]
@hsheth2 I was thinking about this earlier. Do you think we could return an `error`, rather than `panic()` on unsupported platforms (as in `syscalls_other.go`)?

**Comment by hsheth2** [comment]
That would probably be a good idea. I agree that `panic()` is not needed, and that `error` should suffice.

**Comment by songgao** [comment]
@hsheth2 Thanks! I just updated the PR.

**Comment by hsheth2** [comment]
@songgao Looks good to me!

**Pull request merged**